WebSuite.TestGetClusterDetails is flaky

**What happened**: This test was added fairly recently and seems to fail a lot with `list of proxies empty`. Whenever I see unit test failures, it's usually this causing it. Looks racy?

```
FAIL: apiserver_test.go:1640: WebSuite.TestGetClusterDetails

apiserver_test.go:1646:
    c.Assert(err, IsNil)
... value *trace.TraceErr = 
ERROR REPORT:
Original Error: *trace.NotFoundError list of proxies empty
Stack Trace:
	/gopath/src/github.com/gravitational/teleport/lib/services/server.go:786 github.com/gravitational/teleport/lib/services.GuessProxyHostAndVersion
	/gopath/src/github.com/gravitational/teleport/lib/web/ui/cluster.go:82 github.com/gravitational/teleport/lib/web/ui.GetClusterDetails
	/gopath/src/github.com/gravitational/teleport/lib/web/apiserver_test.go:1645 github.com/gravitational/teleport/lib/web.(*WebSuite).TestGetClusterDetails
	/opt/go/src/reflect/value.go:463 reflect.Value.call
	/opt/go/src/reflect/value.go:321 reflect.Value.Call
	/gopath/src/github.com/gravitational/teleport/vendor/gopkg.in/check.v1/check.go:764 github.com/gravitational/teleport/vendor/gopkg.in/check%2ev1.(*suiteRunner).forkTest.func1
	/gopath/src/github.com/gravitational/teleport/vendor/gopkg.in/check.v1/check.go:658 github.com/gravitational/teleport/vendor/gopkg.in/check%2ev1.(*suiteRunner).forkCall.func1
	/opt/go/src/runtime/asm_amd64.s:1358 runtime.goexit
User Message: list of proxies empty
 ("list of proxies empty")

OOPS: 30 passed, 1 FAILED
--- FAIL: TestWeb (16.44s)
```

**What you expected to happen**: Test should pass 100% of the time when there's no actual error.

**How to reproduce it (as minimally and precisely as possible)**: Run tests on Drone, observe failure.

--- a/lib/web/apiserver_test.go
+++ b/lib/web/apiserver_test.go
@@ -244,6 +244,18 @@ func (s *WebSuite) SetUpTest(c *C) {
 	err = s.proxy.Start()
 	c.Assert(err, IsNil)
 
+	// Wait for proxy to fully register before starting the test.
+	for start := time.Now(); ; {
+		proxies, err := s.proxyClient.GetProxies()
+		c.Assert(err, IsNil)
+		if len(proxies) != 0 {
+			break
+		}
+		if time.Since(start) > 5*time.Second {
+			c.Fatal("proxy didn't register within 5s after startup")
+		}
+	}
+
 	proxyAddr := utils.MustParseAddr(s.proxy.Addr())
 
 	addr := utils.MustParseAddr(s.webServer.Listener.Addr().String())

